crates/sui-bridge/src/metrics.rs

```diff
@@ -15,6 +20,140 @@ const FINE_GRAINED_LATENCY_SEC_BUCKETS: &[f64] = &[
     200., 250., 300., 350., 400.,
 ];
 
+pub struct MetricsPushClient {
+    certificate: std::sync::Arc<sui_tls::SelfSignedCertificate>,
+    client: reqwest::Client,
+}
+
+impl MetricsPushClient {
+    pub fn new(metrics_key: sui_types::crypto::NetworkKeyPair) -> Self {
+        use fastcrypto::traits::KeyPair;
+        let certificate = std::sync::Arc::new(sui_tls::SelfSignedCertificate::new(
+            metrics_key.private(),
+            sui_tls::SUI_VALIDATOR_SERVER_NAME,
+        ));
+        let identity = certificate.reqwest_identity();
+        let client = reqwest::Client::builder()
+            .identity(identity)
+            .build()
+            .unwrap();
+
+        Self {
+            certificate,
+            client,
+        }
+    }
+
+    pub fn certificate(&self) -> &sui_tls::SelfSignedCertificate {
+        &self.certificate
+    }
+
+    pub fn client(&self) -> &reqwest::Client {
+        &self.client
+    }
+}
+
+/// Starts a task to periodically push metrics to a configured endpoint if a metrics push endpoint
+/// is configured.
+pub fn start_metrics_push_task(
+    metrics_config: &Option<MetricsConfig>,
+    metrics_key_pair: NetworkKeyPair,
+    registry: RegistryService,
+) {
+    use fastcrypto::traits::KeyPair;
+
+    const DEFAULT_METRICS_PUSH_INTERVAL: Duration = Duration::from_secs(60);
```

Comment:
is this the default value we use for validator node? it seems a bit large

Reply:
It is, from mysten validators configs:
```
metrics:
  push-interval-seconds: 60
  ```